proposals/mdm-baseline.md

```diff
@@ -0,0 +1,153 @@
+# MDM core
+
+This document is a high-level spec/guide of the technical challenges that don't
+directly map to a product feature request.
+
+Some of this might already be captured in other issues, but we wanted to put
+everything together and have multiple people thinking about the solutions as a
+whole.
+
+We will create issues to tackle actionable bits, but the idea is to also use
+this to guide the implementation of feature requests.
+
+
+### Problem: Hosts can be enrolled via MDM
+
+Currently, there's no reliable way to associate an MDM enrolled device with a
+host in the `hosts` table.
+
+- If a host is enrolled in MDM before its enrolled via `osquery` (eg: DEP
+  enrollment) we will have a record in the `nano_enrollments` table before the
+  host exists in the `hosts` table.
+
+- If a host exists in both tables, we could `JOIN` them using `hosts.uuid =
+  nano_enrollments.device_id`  but we need to make sure the right constraints
+  in the application logic exist and the right indexes are created for fast
+  access.
```

Comment:
How is this table populated? Is this something that comes along with nanomdm?

Reply:
yes, that table (among all the `nano_` prefixed tables) is managed by nanomdm. We were thinking on not touching those tables if possible.

Reply:
So nanomdm already handles the ingestion of host data from apple?

Reply:
kind of... (see `nano_devices` and `nano_enrollments`) I will try to clarify this in the spec but you have:

1. Hosts registered in ABM but not enrolled yet.
2. Hosts enrolled through MDM that don't have `osquery` installed yet.
3. Host enrolled through `osquery` that are not enrolled in MDM yet.

we need to list all of them as part of `GET /hosts` and other host-APIs.